test/Data/String/ConversionsSpec.hs

```diff
@@ -10,11 +11,184 @@ import           Data.String.Conversions
+
+      it "converts String -> LazyByteString" $ do
+        property $ \(s :: String) ->
+          deepseq (toLazyByteString s) pass
+
+      it "converts StrictText -> String" $ do
+        property $ \(s :: StrictText) ->
+          deepseq (toString s) pass
+
+      it "converts StrictText -> StrictText" $ do
+        property $ \(s :: StrictText) ->
+          deepseq (toStrictText s) pass
+
+      it "converts StrictText -> LazyText" $ do
+        property $ \(s :: StrictText) ->
+          deepseq (toLazyText s) pass
+
+      it "converts StrictText -> StrictByteString" $ do
+        property $ \(s :: StrictText) ->
+          deepseq (toStrictByteString s) pass
+
+      it "converts StrictText -> LazyByteString" $ do
+        property $ \(s :: StrictText) ->
+          deepseq (toLazyByteString s) pass
+
+      it "converts LazyText -> String" $ do
+        property $ \(s :: LazyText) ->
+          deepseq (toString s) pass
+
+      it "converts LazyText -> StrictText" $ do
+        property $ \(s :: LazyText) ->
+          deepseq (toStrictText s) pass
+
+      it "converts LazyText -> LazyText" $ do
+        property $ \(s :: LazyText) ->
+          deepseq (toLazyText s) pass
+
+      it "converts LazyText -> StrictByteString" $ do
+        property $ \(s :: LazyText) ->
+          deepseq (toStrictByteString s) pass
+
+      it "converts LazyText -> LazyByteString" $ do
+        property $ \(s :: LazyText) ->
+          deepseq (toLazyByteString s) pass
+
+      it "converts StrictByteString -> String" $ do
+        property $ \(s :: StrictByteString) ->
+          deepseq (toString s) pass
+
+      it "converts StrictByteString -> StrictText" $ do
+        property $ \(s :: StrictByteString) ->
+          deepseq (toStrictText s) pass
+
+      it "converts StrictByteString -> LazyText" $ do
+        property $ \(s :: StrictByteString) ->
+          deepseq (toLazyText s) pass
+
+      it "converts StrictByteString -> StrictByteString" $ do
+        property $ \(s :: StrictByteString) ->
+          deepseq (toStrictByteString s) pass
+
+      it "converts StrictByteString -> LazyByteString" $ do
+        property $ \(s :: StrictByteString) ->
+          deepseq (toLazyByteString s) pass
+
+      it "converts LazyByteString -> String" $ do
+        property $ \(s :: LazyByteString) ->
+          deepseq (toString s) pass
+
+      it "converts LazyByteString -> StrictText" $ do
+        property $ \(s :: LazyByteString) ->
+          deepseq (toStrictText s) pass
+
+      it "converts LazyByteString -> LazyText" $ do
+        property $ \(s :: LazyByteString) ->
+          deepseq (toLazyText s) pass
+
+      it "converts LazyByteString -> StrictByteString" $ do
+        property $ \(s :: LazyByteString) ->
+          deepseq (toStrictByteString s) pass
+
+      it "converts LazyByteString -> LazyByteString" $ do
+        property $ \(s :: LazyByteString) ->
+          deepseq (toLazyByteString s) pass
```

Comment:
What do you think about a helper function like this:
``` haskell
smokeTest :: (Arbitrary a, Show a, NFData b) => (a -> b) -> Property
smokeTest convert = do
  property $ \(s :: a) ->
    deepseq (convert s) (return () :: IO ())
```
Then the individual tests would be e.g.:
``` haskell
it "converts LazyByteString -> LazyByteString" $ do
  smokeTest (cs :: LazyByteString -> LazyByteString)
```

Reply:
We could also have a similar helper function for the roundtrip tests above.